### PR TITLE
Refactor the repository

### PR DESCRIPTION
**Pull request by arsulegai**
1. Create internal package
2. Move deployment and asset/static files
3. Introduce Makefile
4. Update readme information

Signed-off-by: S m, Aruna <arun.s.m.cse@gmail.com>

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -18,5 +18,5 @@ generated-data/*.json
 .gitignore
 
 #ignore docker compose file
-docker-compose.yaml
+deployments/docker-compose.yaml
 .DS_Store
```
Comment (arnabkaycee) [inline .dockerignore:21]: [nitpick] - should we add a new line at EOF?

File: .github/workflows/hyperledger-updates.yml [hunk 2]
```diff
@@ -16,10 +16,10 @@ jobs:
...
-        go-version: 1.15
+        go-version: 1.16
 
     - name: Build
-      run: go build -v ./...
+      run: make
 
     - name: Test
-      run: go test -v ./...
+      run: make unit-test
```
Comment (arnabkaycee) [inline .github/workflows/hyperledger-updates.yml:25]: [nitpick] - should we add a new line at EOF?

File: .gitignore [hunk 3]
```diff
@@ -19,8 +19,8 @@ hyperledger-updates
 .idea
 
 # Ignore generated files
-html/generated/*
-generated-data/*.json
+assets/html/generated/*
+assets/generated-data/*.json
 # Ignore .env file
 .env
 .DS_Store
```
Comment (arnabkaycee) [inline .gitignore:26]: [nitpick] - should we add a new line at EOF?

File: Dockerfile [hunk 4]
```diff
@@ -12,11 +12,19 @@
...
-ENV PATH=${PATH}:/app
+RUN make
+
+WORKDIR /appbin
+RUN cp /app/hyperledger-updates /appbin/
+RUN rm -r /app
+
+ENV PATH=${PATH}:/appbin
 
 CMD ["hyperledger-updates"]
```
Comment (arnabkaycee) [inline Dockerfile:30]: [nitpick] - should we add a new line at EOF?

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+	go build -o hyperledger-updates $(LDFLAGS) ./cmd
+
+unit-test:
+	CGO_ENABLED=0 go test -v ./...
+
+lint-check:
+	@[ "$(format_output)" == "" ] || exit -1
+
+format:
+	go fmt ./...
```
Comment (arnabkaycee) [inline Makefile:22]: [nitpick] - should we add a new line at EOF?

File: deployments/docker-compose.yaml [hunk 6]
```diff
@@ -17,11 +17,11 @@ version: "3.6"
...
+      context: ..
       dockerfile: Dockerfile
     container_name: hyperledger-updates
     environment:
       - GITHUB_TOKEN=${GITHUB_TOKEN}
     volumes:
-      - ./html/generated:/app/html/generated
-      - ./generated-data:/app/generated-data
+      - ../assets/html/generated:/appbin/html/generated
+      - ../assets/generated-data:/appbin/generated-data
```
Comment (arnabkaycee) [inline deployments/docker-compose.yaml:31]: [nitpick] - should we add a new line at EOF?

File: internal/pkg/configs/constants.go [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+	ConfigFile = "CONFIG_FILE"
+	// GitHubToken env variable
+	GitHubToken = "GITHUB_TOKEN"
+	// PrSummaryFilePath env variable
+	PrSummaryFilePath = "PR_SUMMARY_FILE_PATH"
+	// ReleaseSummaryFilePath env variable
+	ReleaseSummaryFilePath = "RELEASE_SUMMARY_FILE_PATH"
+	// IssueSummaryFilePath env variable
+	IssueSummaryFilePath = "ISSUE_SUMMARY_FILE_PATH"
+)
```
Comment (arnabkaycee) [inline internal/pkg/configs/constants.go:36]: [nitpick] - should we add a new line at EOF?

File: internal/pkg/configs/structures.go [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+type ReleaseList struct {
+	Repository string                     `json:"repository,omitempty"`
+	Releases   []github.RepositoryRelease `json:"releases,omitempty"`
+}
+
+type IssueList struct {
+	Repository string         `json:"repository,omitempty"`
+	Labels     []string       `json:"labels,omitempty"`
+	Issues     []github.Issue `json:"issues,omitempty"`
+}
```
Comment (arnabkaycee) [inline internal/pkg/configs/structures.go:77]: [nitpick] - should we add a new line at EOF?

File: internal/pkg/utils/report-utils.go [hunk 9]
```diff
@@ -45,3 +45,11 @@ func PrettyPrint(v interface{}, fileName string, templateFile string) error {
...
 	return err
 }
+
+func GetEnvOrDefault(env, defaultValue string) string {
+	value, isPresent := os.LookupEnv(env)
+	if isPresent {
+		return value
+	}
+	return defaultValue
+}
```
Comment (arnabkaycee) [inline internal/pkg/utils/report-utils.go:55]: [nitpick] - should we add a new line at EOF?

File: Makefile [hunk 10]
```diff
@@ -0,0 +1,22 @@
+SHELL := /bin/bash
```
Comment (arnabkaycee) [inline Makefile:1]: Wont build in docker

```
 => ERROR [5/8] RUN make                                                   0.2s
------
 > [5/8] RUN make:
#9 0.206 make: /bin/bash: No such file or directory
#9 0.206 make: *** [Makefile:10: clean] Error 127
#9 0.206 rm -f hyperledger-updates
```

File: .dockerignore (hunk 1 above)
Comment (arsulegai) [reply .dockerignore:21]: There is new line at the EOF

File: .github/workflows/hyperledger-updates.yml (hunk 2 above)
Comment (arsulegai) [reply .github/workflows/hyperledger-updates.yml:25]: There is new line at the EOF. It shall be handled if we see a weird red symbol on the github PR diff.

File: .gitignore (hunk 3 above)
Comment (arsulegai) [reply .gitignore:26]: There is new line at the EOF. It shall be handled if we see a weird red symbol on the github PR diff.

File: Dockerfile (hunk 4 above)
Comment (arsulegai) [reply Dockerfile:30]: There is new line at the EOF. It shall be handled if we see a weird red symbol on the github PR diff.

File: Makefile (hunk 10 above)
Comment (arsulegai) [reply Makefile:1]: Fixed, added bash to the dockerfile

File: Makefile (hunk 5 above)
Comment (arsulegai) [reply Makefile:22]: There is new line at the EOF. It shall be handled if we see a weird red symbol on the github PR diff.

File: deployments/docker-compose.yaml (hunk 6 above)
Comment (arsulegai) [reply deployments/docker-compose.yaml:31]: There is new line at the EOF. It shall be handled if we see a weird red symbol on the github PR diff.

File: internal/pkg/configs/constants.go (hunk 7 above)
Comment (arsulegai) [reply internal/pkg/configs/constants.go:36]: There is new line at the EOF. It shall be handled if we see a weird red symbol on the github PR diff.

File: internal/pkg/configs/structures.go (hunk 8 above)
Comment (arsulegai) [reply internal/pkg/configs/structures.go:77]: There is new line at the EOF. It shall be handled if we see a weird red symbol on the github PR diff.

File: internal/pkg/utils/report-utils.go (hunk 9 above)
Comment (arsulegai) [reply internal/pkg/utils/report-utils.go:55]: There is new line at the EOF. It shall be handled if we see a weird red symbol on the github PR diff.

**Review by arnabkaycee — APPROVED** [review]

**Pull request merged**